### ISSUE TITLE
BigtableInstance: setting 0 to numNodes or deleting it - fails silently

### ISSUE THREAD

**Issue by balaji** [GoogleCloudPlatform/k8s-config-connector#176]
**Describe the bug**
When the `numNodes` section of the BigtableInstance resource contains illegal value (0) or not present at all, the BigtableInstance update is reported as success, but the node count was not changes - implying silent failure.

**ConfigConnector Version**
1.7.1

**To Reproduce**
1. Create a new Bigtable Instance with KCC
2. Edit the cluster's `numNodes` to 0 (or remove it)

Expected:
Error message in the status saying that number of nodes has to be at least 1 (or mandatory)

Actual:
No error message. No change to node count. This was the status after setting `numNodes` to 0.
```
Status:
  Conditions:
    Last Transition Time:  2020-05-12T16:03:41Z
    Message:               The resource is up to date
    Reason:                UpToDate
    Status:                True
    Type:                  Ready
```

*YAML snippets:*

```yaml
apiVersion: bigtable.cnrm.cloud.google.com/v1beta1
kind: BigtableInstance
metadata:
  name: failing-test-instance1
  namespace: bigtable-playground
spec:
  displayName: test
  instanceType: PRODUCTION
  cluster:
  - clusterId: gew-cluster1
    zone: europe-west1-c
    numNodes: 1
```

```yaml
apiVersion: bigtable.cnrm.cloud.google.com/v1beta1
kind: BigtableInstance
metadata:
  name: failing-test-instance1
  namespace: bigtable-playground
spec:
  displayName: test
  instanceType: PRODUCTION
  cluster:
  - clusterId: gew-cluster1
    zone: europe-west1-c
    numNodes: 0

```

**Comment by kibbles-n-bytes** [comment]
Thanks for filing. I was able to reproduce this issue. We'll update this thread once the fix is in place.

**Comment by jcanseco** [comment]
Hi @balaji, this should have been fixed in v1.13.1. Closing this issue now; though feel free to re-open if you have further issues.
